components/audio_player/CMakeLists.txt

```diff
@@ -0,0 +1,4 @@
+idf_component_register(SRCS "audio_player.cpp" "audio_mp3.cpp" "audio_wav.cpp"
+                       INCLUDE_DIRS "include"
+                       REQUIRES libhelix-mp3
```

Comment:
So, If I adjust this like the unity example I get compilation errors, unable to find a libhelix-mp3 include file, mp3dec.h. However, MESSAGE() indicates that the if(CONFIG_AUDIO_PLAYER_ENABLE_MP3) is being visited, the MESSAGE() is printed out.

It's almost like there is some kind of odd dependency issue where an initial pass is generating some makefiles and a second pass is finding them. I can see this in the cmake output from idf.py build.

And libhelix-mp3 is not included in the include search paths when the compilation fails (which explains why it can't find mp3dec.h).

Using v4.4.2 here, maybe this has been fixed on master?


Oh, and oddly enough it works correctly with adding the audio_wav.cpp and audio_mp3.cpp files, that's not an issue, those files are found and built. So it does look like some odd multi pass thing where the CONFIG options aren't present in the first pass (which causes libhelix-mp3 to not be a dependency, which means its include paths aren't added), but are present on the second pass.

I added some MESSAGE() lines, one inside of the CONFIG_ section and one before idf_component_register()

```
cmorgan@Chriss-MBP mp3_demo % idf.py reconfigure
Executing action: reconfigure
Running cmake in directory /Users/cmorgan/projects/esp-box/examples/mp3_demo/build
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -DIDF_TARGET=esp32s3 -DCCACHE_ENABLE=0 /Users/cmorgan/projects/esp-box/examples/mp3_demo"...
-- Found Git: /usr/local/bin/git (found version "2.37.1") 
-- Component directory /Users/cmorgan/projects/esp-box/components/esp-rainmaker does not contain a CMakeLists.txt file. No component will be added
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /Users/cmorgan/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Users/cmorgan/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Users/cmorgan/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32s3
-- *************** requires is        <---------------- (missing here, but see below where its set)
-- Project sdkconfig file /Users/cmorgan/projects/esp-box/examples/mp3_demo/sdkconfig
Loading defaults file /Users/cmorgan/projects/esp-box/examples/mp3_demo/sdkconfig.defaults...
-- Found Python3: /usr/local/Frameworks/Python.framework/Versions/3.9/bin/python3.9 (found version "3.9.13") found components: Interpreter 
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS - Success
-- App "mp3_demo" version: v0.3.0-18-g0ca9c80-dirty
-- Adding linker script /Users/cmorgan/projects/esp-box/examples/mp3_demo/build/esp-idf/esp_system/ld/memory.ld
-- Adding linker script /Users/cmorgan/esp/esp-idf/components/esp_system/ld/esp32s3/sections.ld.in
-- Adding linker script /Users/cmorgan/esp/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.ld
-- Adding linker script /Users/cmorgan/esp/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.api.ld
-- Adding linker script /Users/cmorgan/esp/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.libgcc.ld
-- Adding linker script /Users/cmorgan/esp/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.newlib.ld
-- Adding linker script /Users/cmorgan/esp/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.version.ld
-- Adding linker script /Users/cmorgan/esp/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.newlib-time.ld
-- Adding linker script /Users/cmorgan/esp/esp-idf/components/soc/esp32s3/ld/esp32s3.peripherals.ld
-- IPV6 support is disabled so the asio component will not be built
-- *********************mp3 support enabled           <--------------------------
-- *************** requires is libhelix-mp3                 <----------------
```

Reply:
@tore-espressif I just put a todo comment in the CMakeLists.txt and sticking with that for the time being. When/if its fixed I'll move it back into the if(). I don't understand the build system well yet.